content/src/service/synchronization/streaming/EventStreamProcessor.ts

```diff
@@ -0,0 +1,109 @@
+import { Writable, Readable } from "stream"
+import parallelTransform from "parallel-transform"
+import log4js from "log4js"
+import { DeploymentWithAuditInfo, EntityId } from "dcl-catalyst-commons";
+import { awaitablePipeline, streamMap, streamFilter, mergeStreams } from "@katalyst/content/service/synchronization/streaming/StreamHelper";
+import { Deployment } from "@katalyst/content/service/deployments/DeploymentManager";
+import { ContentServerClient } from "../clients/ContentServerClient";
+import { HistoryDeploymentOptions } from "../EventDeployer";
+import { OnlyNotDeployedFilter } from "./OnlyNotDeployedFilter";
+
+/**
+ * This class processes a given history as a stream, and even makes some of the downloading in parallel.
+ */
+export class EventStreamProcessor {
+
+    private static readonly LOGGER = log4js.getLogger('EventStreamProcessor');
+    private static readonly PARALLEL_DOWNLOAD_WORKERS = 15
+
+    constructor(
+        private readonly checkIfAlreadyDeployed: (entityIds: EntityId[]) => Promise<Map<EntityId, boolean>>,
+        private readonly deploymentBuilder: DeploymentPreparation) { }
+
+    /**
+     * This method takes many deployment streams and tries to deploy them locally.
+     */
+    async processDeployments(deployments: Readable[], options?: HistoryDeploymentOptions) {
+        // Merge the streams from the different servers
+        const merged = mergeStreams(deployments);
+
+        // TODO: Remove on next deployment
+        // Delete information that is not yet excluded from the response
+        const reduceExtraInfo = streamMap(({ entityType, entityId, entityTimestamp, deployedBy, auditInfo }) => {
```

Comment:
I think It would be better to define these transformations as constants outside the function, so it will be clearer they don't require local information to be created.

Reply:
This transformation you are point to is the only one that doesn't have state and wouldn't need to be instantiated on every sync. But it's also going away on the next deployment